File: python/7 kyu/81.py
# https://www.codewars.com/kata/525f039017c7cd0e1a000a26/train/python

# Passed

def palindrome_chain_length(n):
    is_number_palindrome = lambda num: str(num) == str(num)[::-1]
    steps = 0
    while not is_number_palindrome(n):
        n += int(str(n)[::-1])
        steps += 1

    return steps

output = palindrome_chain_length(87)
print(output)

File: nodejs/7 kyu/81.js
// https://www.codewars.com/kata/6584b7cac29ca91dd9124009/train/javascript

// Passed

function convertLojban(lojban) {
  const lojbanToNumber = {
    pa: 1,
    re: 2,
    ci: 3,
    vo: 4,
    mu: 5,
    xa: 6,
    ze: 7,
    bi: 8,
    so: 9,
    no: 0,
  };

  let lojbanDigit = "";
  let lojbanNumber = "";
  for (let idx = 0; idx < lojban.length; idx++) {
    lojbanDigit += lojban[idx];
    if (lojbanDigit in lojbanToNumber) {
      lojbanNumber += String(lojbanToNumber[lojbanDigit]);
      lojbanDigit = "";
    }
  }
  lojbanNumber = Number(lojbanNumber);
  return lojbanNumber;
}

const output = convertLojban("renonore");
console.log(output);